src/agent/rustjail/src/mount.rs

```diff
@@ -265,6 +265,16 @@ pub fn init_rootfs(
         }
     }
 
+    
```

Comment:
Alright. This is a long standing issue that 9p doesn't honor use-after-unlink on open file descriptors (see https://gitlab.com/qemu-project/qemu/-/issues/103). I tried to fix that when I was actively working on QEMU 9p but it required non-trivial changes on both QEMU and guest 9p driver sides, and I didn't manage to have the 9p driver bits merged.

Good news is that things made substantial progress on the kernel side. It might be worth to resume the QEMU patches (Cc current QEMU 9p maintainer @cschoenebeck).